### PR TITLE
Add filtering report by crate name

### PR DESCRIPTION
**Pull request by peterjoel**
Allowing filtering by crate name instead of vulnerability id. For example, today, two different vulnerabilities were published for the `http` crate, and it is more convenient to ignore both, especially since the fix (remove the dependency on that version of `http`) is the same for both.

I did not write a test, because it wasn't clear how to construct a `Database` from a test, and other similar code does not have tests.

### CODE REVIEW

File: src/report.rs [hunk 1]
```diff
@@ -72,6 +74,9 @@ pub struct Settings {
     /// List of advisory IDs to ignore
     pub ignore: Vec<advisory::Id>,
```
Comment (tarcieri) [inline src/report.rs]: Would suggest changing this to disambiguate:

```suggestion
    pub ignored_advisories: Vec<advisory::Id>,
```

**Comment by Shnatsel** [comment]
I feel filtering by name instead of vuln ID would be dangerous because new, more severe vulnerabilities might be silently ignored by such suppression. Essentially it instructs rustsec to ignore all future vulnerabilities in a specific crate, and it's not a thing you'd ever want to do.

Suppressing individual vulnerabilities is a reasonable model because it requires you to review each one and make a conscious decision, just like Rust's `Result` type compared to exceptions.

**Comment by peterjoel** [comment]
What if the suppression was for a particular version or version range?

In my case, it will take a few days (or more) to update the `http` crate and all transitive dependant crates. If new vulnerabilities are being found in the same version, in the meantime, it's quite frustrating having CI breaking. Ultimately, this is about development ergonomics; I want to know about all vulnerabilities, but I want to pick and choose how our development workflow is interrupted.

**Comment by Shnatsel** [comment]
> What if the suppression was for a particular version or version range?

I've considered that, but sadly it's possible for newly discovered vulnerabilities to only affect versions for which you have suppressions.

> it's quite frustrating having CI breaking

Yes, that is absolutely true. Blocking development because of a vulnerability in a dependency is generally not the right thing to do: you don't care if a dependency is vulnerable or not for development purposes. The action that should be taken instead is rebuilding and redeploying all affected production binaries. This implies that running cargo-audit on CI is not a good idea in the first place. The problem is that we have no better place to put it as of right now because there is no version tracking for production binaries yet, but [there is a draft RFC for it](https://github.com/rust-lang/rfcs/pull/2801) which will allow plugging RustSec tooling into workflows that scan production binaries instead.

**Comment by tarcieri** [comment]
Regardless of whether this feature is a good idea, if we do implement it I think the implementation should use the new [`PackageScope`](https://docs.rs/rustsec/0.18.0/rustsec/database/package_scope/struct.PackageScope.html) struct.

I'd suggest opening an issue to discuss this feature if you'd like to see it through, but for now, I don't think this is how it should be implemented.

**Pull request closed without merging**